lesson_10/solid/solid_app/src/main/java/com/codedifferently/lesson10/library/Librarian.java

```diff
@@ -0,0 +1,62 @@
+package com.codedifferently.lesson10.library;
+
+import java.util.Objects;
+import java.util.Set;
+
+import com.codedifferently.lesson10.library.exceptions.LibraryNotSetException;
+import com.codedifferently.lesson10.library.exceptions.WrongLibraryException;
+
+
+  public class Librarian {
```

Comment:
A librarian should be treated like any other patron of the library. That's a hint that you should consider using inheritance to implement this functionality.